test/usersController.test.js

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable no-underscore-dangle */
+const http = require('http');
+const chai = require('chai');
+const express = require('express');
+const chaiHttp = require('chai-http');
+const { OK, INTERNAL_SERVER_ERROR, NOT_FOUND } = require('http-status-codes');
+const expressRoutesRegistrar = require('express-routes-registrar');
+const db = require('../mongodb');
+const { usersRoutes } = require('../routes');
+const { usersController } = require('../controllers');
+const { Users } = require('../models');
+
+const { expect } = chai;
+
+chai.use(chaiHttp);
+
+describe('Users Routes', () => {
+  const PORT = 3000;
+  const URL = `http://localhost:${PORT}`;
+  const app = express();
+  app.use(express.json());
+  const server = http.createServer(app);
+  const registrar = expressRoutesRegistrar(app);
+
+  before((done) => {
+    db.connect()
+      .then(() => server.listen(PORT))
+      .then(() => registrar.registerRoutesJson(
+        usersRoutes,
+        usersController
+      ))
+      .then(() => done());
+  });
+  before((done) => {
+    Users.deleteMany({})
+      .then(() => done());
+  });
+  after((done) => server.close(done));
+
+  describe('Create /users', () => {
+    it('save a new user (name, type)', () => (
+      chai
+        .request(URL)
+        .post('/users')
+        .send({ name: 'LOL', type: 'employee' })
+        .then(({ body, statusCode }) => {
+          expect(body).to.be.an('object');
+          expect(statusCode).to.be.equal(OK);
+        })
+    ));
+    it('error to save a new user without parameters', () => (
+      chai
+        .request(URL)
+        .post('/users')
+        .send({ })
+        .then(({ statusCode, body: { message: { _message } } }) => {
+          expect(_message).to.be.equal('User validation failed');
+          expect(statusCode).to.be.equal(INTERNAL_SERVER_ERROR);
+        })
+    ));
+  });
+
+  describe('Read /users', () => {
+    it('should return json as listo of users', () => (
+      chai
+        .request(URL)
+        .get('/users')
+        .then(({ body, statusCode }) => {
+          expect(body).to.be.an('array');
+          expect(statusCode).to.be.equal(OK);
+        })
+    ));
+  });
+  describe('Read /users/:id', () => {
+    it('should return an user by id', () => (
+      Users.findOne()
```

Comment:
You can use the before hook to obtaine the Id and then use the same id for all your test cases https://mochajs.org/#hooks

```js
let ID = null;

describe(..., () => {
  before(() => (
    Users.findOne().then(({ _id}) => ID = _id)
  ));
// then your test cases can be smaller

        request(URL)
          .get(`/users/${id}`))
          .then(({ body, statusCode }) => {
              expect(body).to.be.an('object');
              expect(statusCode).to.be.equal(OK);
          })

})
```